decor/sniff: add detection for Windows Phone

AdrianVasiliu opened the pull request:
decor/sniff detects already iOS and Android, but doesn't detect the third supported mobile platform, Windows Phone. This would be necessary for instance for Combobox' support of Windows Phone.

Remarks:
* The name "wp" for the has-feature has been picked from a list of other candidates: "WindowsPhone" (seems we should stick to lower-case), "windowsphone" (unreadable), "windows-phone" (not bad, but longer; found at least one occurrence of dash separator in has-features in requirejs-dplugins; not sure we have explicite guidelines about naming pattern for has-features). @wkeese, if you prefer another name than the short "wp", just tell.
* Differently than the existing detection of "ios" and "android", I pllaced the detection for "wp" outside the webkit detection, for the reason see #30. 
* UA doc on WP: http://msdn.microsoft.com/en-us/library/ie/hh869301(v=vs.85).aspx
* Tested on both WP 8.0 and 8.1.

AdrianVasiliu commented on sniff.js:
```diff
@@ -65,6 +65,11 @@ define(["./features"], function (has) {
+		var wp = parseFloat(dua.split("Windows Phone ")[1]) || undefined;
+		if (wp) {
+			has.add("wp", wp);
+		}
```
Could be simplified to just one line:
```
has.add("wp", parseFloat(dua.split("Windows Phone ")[1]) || undefined);
```
but this would lead to an unnecessary entry for "wp" in has' cache when not running on WP, given the current implementation of has.add (https://github.com/ibm-js/requirejs-dplugins/blob/0.4.2/has.js#L54-59). It looks that has.add could avoid adding the entry if the value is undefined. @clmath?

cjolif replied:
Ultimately we should be consistent and share the same approach across the has values (just adobe mac & msapp are using a different strategy)

AdrianVasiliu replied:
On a related note, see also https://github.com/ibm-js/decor/commit/377223f16810dbbe266b3dbbcc26c0d72fe44191#commitcomment-9274368.

clmath replied:
> It looks that has.add could avoid adding the entry if the value is undefined. @clmath?

Yes it could. OTOH it will add complexity to the has plugin and this is far from being a performance issue.
So I would rather stay with the current implementation but if other people think this should be done I could do it.

wkeese replied:
@cjolif wrote:

> Ultimately we should be consistent and share the same approach across the has values

So then, change it to @AdrianVasiliu's suggestion of this?

```js
has.add("wp", parseFloat(dua.split("Windows Phone ")[1]) || undefined);
```

wkeese commented:
I had actually forgotten that we said we'd support Windows Phone, but anyway "wp" sounds like a fine name to me.

wkeese commented:
PS: I fixed your #30, and checked in a manual test file for sniff; so you should add wp there.

AdrianVasiliu commented:
>  I fixed your #30, and checked in a manual test file for sniff; so you should add wp there.

Ok, added test the testing (and made detection of wp similar to others) via
8d2f141871e76efd51bd5d24e5a81c70b6fa2b61

If no objection, I'll squash and push.

BTW, in terms of testing, more testing of sniff might come via https://github.com/ibm-js/decor/issues/24#issuecomment-64814336 .

wkeese commented:
Sounds good to me.

AdrianVasiliu commented:
Pushed via c3f9cdefec57ba17881b7dbca0776adfe57647c4.
